packages/expo-cli/src/commands/upload/submission-service/utils/files.ts

```diff
@@ -23,4 +81,46 @@ async function uploadAppArchiveAsync(path: string): Promise<string> {
   );
 }
 
-export { downloadAppArchiveAsync, uploadAppArchiveAsync };
+async function createTemporaryDirectoryForExtractionAsync(): Promise<string> {
+  // Since we may need to rename the destination path,
+  // add everything to a folder which can be nuked to ensure we don't accidentally use an old build with the same name.
+  const destinationFolder = join(os.tmpdir(), 'expo-submission-service');
```

Comment:
@EvanBacon with this change, only one single process can upload one build artefact at a time, since otherwise the file will be deleted from under that process. E.g. trying to run `expo upload:ios` and `expo upload:android` in parallel will have one of them fail with either:

> Error setting value '/tmp/expo-submission-service/foo-xxxxxx-signed.aab' for option 'aab'
> Could not find aab file at path '/tmp/expo-submission-service/foo-xxxxxx-signed.aab'

or:

> Failed to upload the standalone app to the app store.
> Could not find file at path '/tmp/expo-submission-service/foo-xxxxxx-archive.ipa'

We have build scripts that submits several builds at the same time, and this is currently breaking our workflow. Would it be possible to use a new folder each time, instead of always using `expo-submission-service`?

I'll send a pull request with that change, but I'm also happy to consider any other approach to solve this 👍